api/main.go

```diff
@@ -77,545 +76,707 @@ func NewApiServer(logger *zap.SugaredLogger, address string, metaAddr string, sr
 	}, nil
 }
 
-func (serv *MetadataServer) CreateUser(ctx context.Context, user *pb.User) (*pb.Empty, error) {
-	serv.Logger.Infow("Creating User", "user", user.Name)
-	return serv.meta.CreateUser(ctx, user)
+func (serv *MetadataServer) CreateUser(ctx context.Context, userRequest *pb.UserRequest) (*pb.Empty, error) {
+	requestID, ctx, logger := serv.Logger.InitializeRequestID(ctx)
+	logger = logger.WithResource("user", userRequest.User.Name, "")
```

Comment:
Ok I know this is annoying but could you actually make a "NoVariant" const and not add the variant to the log as well